test/e2e/e2ev2.go

```diff
@@ -0,0 +1,255 @@
+			resources, err := kbTest.RunKustomizeCommand("build", filepath.Join("config", "default"))
+			if err != nil {
+				fmt.Fprintf(GinkgoWriter, "error when running kustomize build during cleaning up: %v\n", err)
+			}
+			if _, err = kbTest.RunKubectlCommandWithInput(resources, "delete", "--recursive", "-f", "-"); err != nil {
+				fmt.Fprintf(GinkgoWriter, "error when running kubectl delete during cleaning up api objects: %v\n", err)
+			}
+			if _, err = kbTest.RunKubectlCommand(
+				"delete", "--recursive",
+				"-f", filepath.Join("config", "crds"),
+			); err != nil {
+				fmt.Fprintf(GinkgoWriter, "error when running kubectl delete during cleaning up crd: %v\n", err)
+			}
+
+			By("uninstalling cert manager bundle")
+			if _, err = kbTest.RunKubectlDelete(false, "-f", "https://github.com/jetstack/cert-manager/releases/download/v0.8.0/cert-manager.yaml"); err != nil {
+				fmt.Fprintf(GinkgoWriter, "error when running kubectl delete during cleaning up cert manager: %v\n", err)
+			}
+			if _, err = kbTest.RunKubectlDelete(false, "namespace", "cert-manager"); err != nil {
+				fmt.Fprintf(GinkgoWriter, "error when running kubectl delete to cleanup cert manager namespace: %v\n", err)
+			}
+
+			By("remove container image created during test")
+			if err = kbTest.CleanupImage(c.controllerImageName); err != nil {
+				fmt.Fprintf(GinkgoWriter, "error when removing the local image: %v\n", err)
+			}
+
+			By("remove test work dir")
+			if err = os.RemoveAll(c.workDir); err != nil {
+				fmt.Fprintf(GinkgoWriter, "error when removing the test work dir: %v\n", err)
+			}
+		})
+
+		It("should generate a runnable project", func() {
+			var controllerPodName string
+
+			// It takes some time for cert-manager to come up, so do it first to save some time.
+			By("installing cert manager bundle")
+			_, err := kbTest.RunKubectlCommand("create", "namespace", "cert-manager")
+			Expect(err).Should(Succeed())
+			_, err = kbTest.RunKubectlCommand("label", "namespace", "cert-manager", "certmanager.k8s.io/disable-validation=true")
+			Expect(err).Should(Succeed())
+			_, err = kbTest.RunKubectlApply(false, "-f", "https://github.com/jetstack/cert-manager/releases/download/v0.8.0/cert-manager.yaml")
+			Expect(err).Should(Succeed())
+
+			By("init v1 project")
+			err = kbTest.Init(
+				"--project-version", "2",
+				"--domain", c.domain,
+				"--dep=false")
+			Expect(err).Should(Succeed())
+
+			By("creating api definition")
+			err = kbTest.CreateAPI(
+				"--group", c.group,
+				"--version", c.version,
+				"--kind", c.kind,
+				"--namespaced",
+				"--resource",
+				"--controller",
+				"--make=false")
+			Expect(err).Should(Succeed())
+
+			By("implementing the API")
+			err = insertCode(
+				filepath.Join(c.workDir, "api", c.version, fmt.Sprintf("%s_types.go", strings.ToLower(c.kind))),
+				"type "+c.kind+"Spec struct {\n",
+				"\t// +optional\n\tCount int "+"`"+`json:"count,omitempty"`+"`\n")
```

Comment:
We can use multiline literal syntax to improve the readability ?

Reply:
Not sure how much better it is now.